pkg/controller/common/cleaner.go

```diff
@@ -45,7 +45,7 @@ func NewCleaner(manager ctrl.Manager, name controllername.ControllerNamer, gvrRe
 		return fmt.Errorf("creating clientset: %w", err)
 	}
 
-	mapper, err := apiutil.NewDynamicRESTMapper(manager.GetConfig(), apiutil.WithLazyDiscovery)
+	mapper, err := apiutil.NewDynamicRESTMapper(manager.GetConfig(), manager.GetHTTPClient())
```

Comment:
does the http client still use lazy discovery?

Worried about how many calls the rest mapper might make initially 

https://github.com/kubernetes-sigs/controller-runtime/pull/2116

Reply:
lazy by default apparently -- that entire block of code is removed
https://github.com/kubernetes-sigs/controller-runtime/blob/4332e3a54f000d642f3737f43d302ea6c5a91895/pkg/client/apiutil/restmapper_test.go#L84